packages/@aws-cdk/cx-api/lib/cloud-assembly.ts

```diff
@@ -257,65 +227,71 @@ export class CloudAssemblyBuilder {
    * @param options
    */
   public buildAssembly(options: AssemblyBuildOptions = { }): CloudAssembly {
-    const manifest: AssemblyManifest = filterUndefined({
-      version: CLOUD_ASSEMBLY_VERSION,
+
+    // explicitly initializing this type will help us detect
+    // breaking changes. (For example adding a required property will break compilation).
+    let manifest: cxschema.AssemblyManifest = {
+      version: cxschema.Manifest.version(),
       artifacts: this.artifacts,
       runtime: options.runtimeInfo,
       missing: this.missing.length > 0 ? this.missing : undefined
-    });
+    };
+
+    // now we can filter
+    manifest = filterUndefined(manifest);
 
     const manifestFilePath = path.join(this.outdir, MANIFEST_FILE);
-    fs.writeFileSync(manifestFilePath, JSON.stringify(manifest, undefined, 2));
+    cxschema.Manifest.save(manifest, manifestFilePath);
 
     // "backwards compatibility": in order for the old CLI to tell the user they
     // need a new version, we'll emit the legacy manifest with only "version".
     // this will result in an error "CDK Toolkit >= CLOUD_ASSEMBLY_VERSION is required in order to interact with this program."
-    fs.writeFileSync(path.join(this.outdir, 'cdk.out'), JSON.stringify({ version: CLOUD_ASSEMBLY_VERSION }));
+    fs.writeFileSync(path.join(this.outdir, 'cdk.out'), JSON.stringify({ version: manifest.version }));
 
     return new CloudAssembly(this.outdir);
   }
+
 }
 
-export interface AssemblyBuildOptions {
-  /**
-   * Include the specified runtime information (module versions) in manifest.
-   * @default - if this option is not specified, runtime info will not be included
-   */
-  readonly runtimeInfo?: RuntimeInfo;
+/**
+ * Backwards compatibility for when `RuntimeInfo`
+ * was defined here. This is necessary because its used as an input in the stable
+ * @aws-cdk/core library.
+ *
+ * @see core.ConstructNode.synth
```

Comment:
Add "@deprecated" notice on all of them.

Reply:
👍